internal/goldendataset/span_generator.go

```diff
@@ -0,0 +1,559 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goldendataset
+
+import (
+	"crypto/rand"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	otlpcommon "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+)
+
+const (
+	ColumnParent     = 0
+	ColumnTracestate = 1
+	ColumnKind       = 2
+	ColumnAttributes = 3
+	ColumnEvents     = 4
+	ColumnLinks      = 5
+	ColumnStatus     = 6
+)
+
+const (
```

Comment:
I think all these const sections will benefit it each had a corresponding type alias that declares a strong type to use in these consts and later in the code instead of using `string` for everything. With everything being `string` it is easy to make accidental mistakes.

Reply:
Added types